### PR TITLE
Timetable period changes

### PR DESCRIPTION
**Pull request by Jenita**
Added New table aPeriod and modified timetable procedure and substitute teacher

### CODE REVIEW

File: server/database/admin/query.ts [hunk 1]
```diff
@@ -1233,3 +1235,25 @@ export const socialAssessmentMarksQuery: socialAssessmentMarksQueryInterface = {
...
     },
 };
+interface PeriodQueryInterface {
+    createPeriod: (data: Period) => string;
+    updatePeriod: (data: Period) => string;
+}
+
+export const periodQuery: PeriodQueryInterface = {
+    createPeriod: (data: Period) => {
+        const query: any = `INSERT INTO Period( school_id,name,academic_year,start_time,end_time, created_time, created_by,active) VALUES
```
Comment (ArunprasadRajendran) [inline server/database/admin/query.ts:1245]: Pls pull the latest changes from develop branch and use escapeRequestBody method for create and update flow.
refer other flow and follow the same

File: server/database/admin/query.ts [hunk 2]
```diff
@@ -1233,3 +1235,25 @@ export const socialAssessmentMarksQuery: socialAssessmentMarksQueryInterface = {
...
+         (${data.school_id},'${data.name}','${data.academic_year}','${data.start_time}','${data.end_time}','${data.created_time}',${data.created_by},1)`;
+        return query;
+    },
+    updatePeriod: (data: Period) => {
+        const stringData: any = JSON.stringify(data, (key: any, value: any) => {
+            if (value === null) {
+                return;
+            }
+            return value;
+        });
```
Comment (ArunprasadRajendran) [inline server/database/admin/query.ts:1255]: const stringData: any = JSON.stringify(data, (key: any, value: any) => {
            if (value === null) {
                return;
            }
            return value;
        });
        
use a common method for the code change. refer other stored procedure call

**Review by ArunprasadRajendran — CHANGES_REQUESTED** [review]

File: server/database/admin/query.ts (hunk 1 above)
Comment (Jenita) [reply server/database/admin/query.ts:1246]: Updated

File: server/database/admin/query.ts (hunk 2 above)
Comment (Jenita) [reply server/database/admin/query.ts]: Updated

**Review by ArunprasadRajendran — APPROVED** [review]

**Review by ArunprasadRajendran — APPROVED** [review]

**Pull request merged**